fpga_interchange/arch.cc

```diff
@@ -573,4 +693,95 @@ const std::vector<std::string> Arch::availablePlacers = {"sa",
 const std::string Arch::defaultRouter = "router2";
 const std::vector<std::string> Arch::availableRouters = {"router1", "router2"};
 
+void Arch::map_cell_pins(CellInfo *cell, int32_t mapping) const
+{
+    cell->cell_mapping = mapping;
+    cell->pins.clear();
+    cell->cell_bel_pins.clear();
+
+    const CellBelMapPOD &cell_pin_map = chip_info->cell_map->cell_bel_map[mapping];
+
+    log_info("Mapping cell %s\n", cell->name.c_str(this));
+    for (const auto &pin_map : cell_pin_map.common_pins) {
+        IdString cell_pin(pin_map.cell_pin);
+        IdString bel_pin(pin_map.bel_pin);
+
+        if(cell_pin.str(this) == "GND") {
+            // FIXME: Tie this pin to the GND net
+            continue;
+        }
+        if(cell_pin.str(this) == "VCC") {
+            // FIXME: Tie this pin to the VCC net
+            continue;
+        }
+
+        auto &bel_pins = cell->cell_bel_pins[cell_pin];
+        log_info("%s => %s\n", cell_pin.c_str(this), bel_pin.c_str(this));
+        cell->pins[cell_pin] = bel_pin;
+        bel_pins.push_back(bel_pin);
+
+        // FIXME: Only have support for 1-to-1 pin in the router right now.
```

Comment:
in case you haven't seen it, #575

Reply:
Right.  For now, it is probably best that I just debug 1-to-1 mappings (e.g. LUT's, FF's, LUT-RAM's, CARRY's). Just getting to the point where I can place and route those items, this arch will be in a good starting place!

Reply:
Yes that's sensible

Reply:
This was handled in #586